playwright/src/main/java/com/microsoft/playwright/junit/impl/APIRequestContextExtension.java

```diff
@@ -7,9 +7,14 @@
 
 import static com.microsoft.playwright.junit.impl.ExtensionUtils.isParameterSupported;
 
-public class APIRequestContextExtension implements ParameterResolver, AfterAllCallback {
+public class APIRequestContextExtension implements ParameterResolver, BeforeEachCallback, AfterAllCallback {
   private static final ThreadLocal<APIRequestContext> threadLocalAPIRequestContext = new ThreadLocal<>();
 
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) {
```

Comment:
I'd not do this and keep it in `afterEach` only. Otherwise we might introduce a fixture that uses APIRequestContext (in its beforeEach) which is later removed and the test may end up using another one, different from the beforeEach fixture. Alternatively, we could store APIRequestContext created for beforeAll in a separate thread local and clear only that one in beforeEach and have the test's one cleared in afteEach/

It looks like it could be easier to not provide APIRequestContext in `beforeAll` for now. WDYT?

Reply:
this beforeEach is execute before any user-defined beforeEach, so it's not possible for this beforeEach to remove before the test is executed.  Unless i'm misunderstanding your concern here.

Reply:
You understood it correctly. Thanks for explaining, let's try it then.